### PR TITLE
Update matrix usage limits

### PR DESCRIPTION
**Pull request by ChristopherMank**
This actions reusable is used in the docs for both GitHub-hosted and self-hosted runners. Given that, the language should support that.

### Why:

Closes [issue link]

### Check off the following:

- [x] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by ChristopherMank** [comment]
I reviewed the contributing guide and given this was a quick fix, I did not open an issue for it (as it states). Please let me know if I should do that for every case.

**Comment by ramyaparimi** [comment]
@ChristopherMank 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: data/reusables/actions/usage-matrix-limits.md [hunk 1]
```diff
@@ -1 +1 @@
-A job matrix can generate a maximum of 256 jobs per workflow run. This limit also applies to self-hosted runners.
+A job matrix can generate a maximum of 256 jobs per workflow run. This limit applies to both GitHub-hosted and self-hosted runners.
```
Comment (skedwards88) [inline data/reusables/actions/usage-matrix-limits.md:1]:
```suggestion
A job matrix can generate a maximum of 256 jobs per workflow run. This limit applies to both {% data variables.product.product_name %}-hosted and self-hosted runners.
```

**Review by skedwards88 — COMMENTED** [review]
Thanks for this update! I'm going to make one minor change to use a variable, then we'll get this merged down for you.

**Review by skedwards88 — APPROVED** [review]

**Pull request merged**